### PR TITLE
Ken/upgrade expo

### PR DESCRIPTION
**Pull request by lbkchen**
* Upgraded Expo 15.0.0 -> 31.0.0
* Upgraded RN, React to latest versions
* Upgraded React Navigation and refactor to fix bugs
* Putting on hold until we have token auth
* Have a formatting plugin so a lot of files just got formatted

@JasonBaoZ How were you able to even load the home page last semester?

In `veterans_controller.rb` we need the `current_veteran` object but that doesn't get initialized until the veteran gets authenticated? 

```ruby
  def index
    @veterans = Veteran.all
    respond_to do |format|
      format.html { render :index }
      format.json {
        render json: @veterans,
               each_serializer: VeteranFriendSerializer,
               scope: { current_veteran: current_veteran }
      }
    end
  end
```

### CODE REVIEW

File: src/components/ConnectBox.js [hunk 1]
```diff
@@ -53,7 +46,7 @@ export default class ConnectBox extends React.Component {
   }
 
   componentDidMount() {
-    Animations.fade(this.state.animationValue, toValue = 1).start();
+    Animations.fade(this.state.animationValue, (toValue = 1)).start();
```
Comment (jasonzbao) [inline src/components/ConnectBox.js:49]: Just curious, what does this do?

**Review by jasonzbao — APPROVED** [review]
LGTM most of this is just styling right?

**Comment by jasonzbao** [comment]
By the home page do you mean the web home page or the mobile? @lbkchen

File: src/components/ConnectBox.js (hunk 1 above)
Comment (lbkchen) [reply src/components/ConnectBox.js:49]: nothing! you're right that a lot of this is styling cause I added an auto-linter that formats files for me. 

no functional change here

File: src/navigators/ConnectSignUpStackNavigator.js [hunk 2]
```diff
@@ -3,27 +3,29 @@
...
+import { createStackNavigator } from "react-navigation";
+import ConnectScreen from "../screens/ConnectScreen";
+import ConnectSignUpScreen from "../screens/ConnectSignUpScreen";
+import ProfileScreen from "../screens/ProfileScreen";
 
-const ConnectSignUpStackNavigator = StackNavigator({
-  Connect: {
-    screen: ConnectScreen,
+const ConnectSignUpStackNavigator = createStackNavigator(
+  {
```
Comment (lbkchen) [inline src/navigators/ConnectSignUpStackNavigator.js:14]: This is one of the main changes though cause newer react navigation usage is pretty different from how we did things before

File: src/screens/LoginScreen.js [hunk 3]
```diff
@@ -108,15 +101,15 @@ export default class LoginScreen extends React.Component {
    * Routes the user to the app.
    */
   navigateToApp(navProps) {
-    this.props.navigation.navigate('App', navProps);
+    this.props.navigation.navigate("Home", navProps);
```
Comment (lbkchen) [inline src/screens/LoginScreen.js:104]: this was a bug in the old version--the navprops don't get passed to the screen if you use `App`, use `Home` instead

**Comment by lbkchen** [comment]
@JasonBaoZ mobile home page! web seems to work for me

File: src/screens/LoginScreen.js (hunk 3 above)
Comment (jasonzbao) [reply src/screens/LoginScreen.js:104]: Lol I have no idea how it loaded :'(

**Pull request merged**